public/app/features/plugins/pluginSettings.test.ts

```diff
@@ -0,0 +1,154 @@
+import { getPluginSettings, clearPluginSettingsCache } from './pluginSettings';
+import { getBackendSrv } from '@grafana/runtime';
+
+jest.mock('@grafana/runtime', () => ({
+  getBackendSrv: jest.fn().mockReturnValue({
+    get: jest.fn(),
+  }),
+}));
+
+describe('PluginSettings', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    clearPluginSettingsCache();
+  });
+
+  it('should fetch settings when cache is empty', async () => {
+    // assign
+    getBackendSrv().get = jest.fn().mockResolvedValue({
+      name: 'TestPlugin',
+      type: 'datasource',
+      id: 'test-plugin',
+      enabled: true,
+    });
+    const getRequestSpy = jest.spyOn(getBackendSrv(), 'get');
+    // act
+    const response = await getPluginSettings('test');
+    // assert
+    expect(response).toEqual({
+      name: 'TestPlugin',
+      type: 'datasource',
+      id: 'test-plugin',
+      enabled: true,
+    });
+    expect(getRequestSpy).toHaveBeenCalledTimes(1);
+    expect(getRequestSpy).toHaveBeenCalledWith('/api/plugins/test/settings');
+  });
+
+  it('should fetch settings from cache when it has a hit', async () => {
+    // assign
+    getBackendSrv().get = jest.fn().mockResolvedValue({
+      name: 'TestPlugin',
+      type: 'datasource',
+      id: 'test-plugin',
+      enabled: true,
+    });
+    const getRequestSpy = jest.spyOn(getBackendSrv(), 'get');
+    // act
+    const response1 = await getPluginSettings('test');
+    const response2 = await getPluginSettings('test');
+
+    // assert
+    expect(response1).toEqual({
+      name: 'TestPlugin',
+      type: 'datasource',
+      id: 'test-plugin',
+      enabled: true,
+    });
+    expect(response2).toEqual({
+      name: 'TestPlugin',
+      type: 'datasource',
+      id: 'test-plugin',
+      enabled: true,
+    });
```

Comment:
**Suggestion:** maybe we could extract the object into a variable to make it a bit more DRY

```javascript
const pluginSettings = {
  name: 'TestPlugin',
  type: 'datasource',
  id: 'test-plugin',
  enabled: true,
};

// ...

expect(response1).toEqual(pluginSettings);
expect(response2).toEqual(pluginSettings);
```

Reply:
@leventebalogh I have made it a bit more cleaner now, thanks. I still like to avoid making the test "arrangements" outside of the test - like pushing this variable outside of the test and re-use it everywhere. This would make the unit tests coupled to each other.

Reply:
> like pushing this variable outside of the test and re-use it everywhere. This would make the unit tests coupled to each other.

Yea, I absolutely agree on that.
It's also easier-to-read this way (one doesn't always have to scroll back to the top to see what that variable is).